sdks/python/apache_beam/options/pipeline_options.py

```diff
@@ -217,7 +217,7 @@ def get_all_options(self, drop_default=False):
     result = vars(known_args)
 
     # Apply the overrides if any
-    for k in result.keys():
+    for k in list(result):
```

Comment:
I don't think we need the `list()` call here.

Reply:
I'm following [http://python-future.org/compatible_idioms.html#dict-keys-values-items-as-a-list](this rule) without which there will be `RuntimeError: dictionary changed size during iteration`

Reply:
You're right, I mist the `del result[k]`.

Reply:
You're right, I missed the `del result[k]`.